### PR TITLE
Initrd fixes for ubuntu systemd

### PR DESCRIPTION
**Pull request by ryansavino**
osbuilder: fix ubuntu initrd /dev/ttyS0 hang
osbuilder: add systemd symlinks for kata-agent

Fixes: [#4932](https://github.com/kata-containers/kata-containers/issues/4932)

Signed-Off-By: Ryan Savino <ryan.savino@amd.com>

### CODE REVIEW

**Review by fidencio — CHANGES_REQUESTED** [review]
@ryansavino, please, could you explain in each one of the commits, in the commit message, why you're doing what you're doing there?  It'll help immensely whoever has to bisect the code in the future.

**Comment by ryansavino** [comment]
> @ryansavino, please, could you explain in each one of the commits, in the commit message, why you're doing what you're doing there? It'll help immensely whoever has to bisect the code in the future.

My bad. I've updated the commit messages, and I think they're more clear now.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @ryansavino!

File: tools/osbuilder/rootfs-builder/rootfs.sh [hunk 1]
```diff
@@ -597,7 +597,13 @@ EOF
 	[ -x "${AGENT_DEST}" ] || die "${AGENT_DEST} is not installed in ${ROOTFS_DIR}"
 	OK "Agent installed"
 
-	[ "${AGENT_INIT}" == "yes" ] && setup_agent_init "${AGENT_DEST}" "${init}"
+	if [ "${AGENT_INIT}" == "yes" ]; then
+		setup_agent_init "${AGENT_DEST}" "${init}"
+	else
+		# Setup systemd service for kata-agent
+		mkdir -p "${ROOTFS_DIR}/etc/systemd/system/basic.target.wants"
+		ln -sf "/usr/lib/systemd/system/kata-containers.target" "${ROOTFS_DIR}/etc/systemd/system/basic.target.wants/kata-containers.target"
```
Comment (amshinde) [inline tools/osbuilder/rootfs-builder/rootfs.sh:610]: Does this have to be "{ROOTFS_DIR}/usr/lib/systemd/system/kata-containers.target" ?

**Comment by amshinde** [comment]
You might also want to update the doc here to reflect the changes you are making:
https://github.com/kata-containers/documentation/wiki/Kata-images

File: tools/osbuilder/rootfs-builder/rootfs.sh (hunk 1 above)
Comment (ryansavino) [reply tools/osbuilder/rootfs-builder/rootfs.sh:610]: This is the same way it's done for the dracut overlay:

https://github.com/kata-containers/kata-containers/blob/main/tools/osbuilder/rootfs-builder/rootfs.sh#L459

The agent is setting it here:

**Comment by ryansavino** [comment]
> You might also want to update the doc here to reflect the changes you are making: https://github.com/kata-containers/documentation/wiki/Kata-images

The documentation repo is showing this:
"This repository has been archived by the owner. It is now read-only."

Is there another place this documentation should be added?

File: tools/osbuilder/rootfs-builder/rootfs.sh (hunk 1 above)
Comment (amshinde) [reply tools/osbuilder/rootfs-builder/rootfs.sh:610]: Never mind, makes sense to me how it is currently. Should be "target" itself.

**Review by amshinde — APPROVED** [review]

**Pull request merged**